### ISSUE TITLE
Error generating primary key

### ISSUE THREAD

**Issue by alienfs** [Impetus/kundera#884]
My code is throwing this error:
15:34:55.361 [http-nio-8080-exec-2] ERROR c.i.k.c.cassandra.dsdriver.DSClient - Error while executing query Select now() from system_schema.columns.
sep 20, 2016 3:34:55 PM org.apache.catalina.core.StandardWrapperValve invoke
SEVERE: Servlet.service() for servlet [Jersey REST Service] in context with path [/WonderING] threw exception [com.impetus.kundera.KunderaException: com.impetus.kundera.KunderaException: com.datastax.driver.core.exceptions.InvalidQueryException: Keyspace system_schema does not exist] with root cause
com.datastax.driver.core.exceptions.InvalidQueryException: Keyspace system_schema does not exist

I found the offending line:
Y:\Kundera-trunk\src\kundera-cassandra\cassandra-core\src\main\java\com\impetus\client\cassandra\CassandraIdGenerator.java (1 hit)
    Line 70:         CqlResult cqlResult = ((CassandraClientBase) client).execute("SELECT NOW() FROM system_schema.columns LIMIT 1",

I checked the list of keyspaces via Datastax DevCenter and I do see a keyspace "system" with table "schema_columns".
I suspect that the query is wrong:
SELECT NOW() FROM system_schema.columns LIMIT 1
--> SELECT NOW() FROM system.schema_columns LIMIT 1

**Comment by karthikprasad13** [comment]
@alienfs 

I believe you are using old version of Cassandra. Please update with the versions of Cassandra and Kundera you are working with.

Compatible versions:
- Cassandra 2.x and Kundera 3.4
- Cassandra 3.x and Kundera 3.5+

-Karthik

**Comment by alienfs** [comment]
Hello Karthik,

Thanks for the quick update!

Indeed, I am connecting to a Cassandra 2,x cluster and was using kundera-cassandra-ds-driver 3.5. I did try to force support via:
Map<String, String> props = new HashMap<>();
props.put(CassandraConstants.CQL_VERSION, CassandraConstants.CQL_VERSION_2_0);
EntityManagerFactory emf = Persistence.createEntityManagerFactory("WonderING_Datastore", props);

I now switched to kundera-cassandra-ds-driver 3.4 and still failed to persist my classes. It seems that auto generation on an int is unsupported. After switching to String, it worked.

@Id
@GeneratedValue(strategy = GenerationType.AUTO)
@Column(name = "id")
private int id;
gives:
javax.servlet.ServletException: com.impetus.kundera.KunderaException: com.impetus.kundera.property.PropertyAccessException: java.lang.NumberFormatException: For input string: "844d9d60-7feb-11e6-a30a-bb19ec7e3ca6"

@Id
@GeneratedValue(strategy = GenerationType.SEQUENCE)
@Column(name = "id")
private int id;
gives:
java.lang.IllegalArgumentException: GenerationType.SEQUENCE Strategy not supported by this client :com.impetus.kundera.client.cassandra.dsdriver.DSClient

@Id
@GeneratedValue(strategy = GenerationType.TABLE)
@Column(name = "id")
private int id;
gives:
[com.impetus.kundera.configure.schema.SchemaGenerationException: com.impetus.kundera.configure.schema.SchemaGenerationException: Error occurred while creating indexes on column sequence_value of table kundera_sequences] with root cause
InvalidRequestException(why:Secondary indexes are not supported on counter tables)

@Id
@GeneratedValue(strategy = GenerationType.IDENTITY)
@Column(name = "id")
private int id;
gives:
java.lang.UnsupportedOperationException: GenerationType.IDENTITY Strategy not supported by this client :com.impetus.kundera.client.cassandra.dsdriver.DSClient

It seems that GenerationType.AUTO uses a different mechanism then the ones I can manually select?

I used int id because the existing data model comes from an Oracle-database, but I don't really have a reason for it to be an int any longer, so switching to String id is fine. For me the issue is solved by switching to an earlier version of kundera-cassandra-ds-driver.
- AlienFS
